specification/mariadb/resource-manager/Microsoft.DBforMariaDB/stable/2018-06-01/PerformanceRecommendations.json

```diff
@@ -0,0 +1,555 @@
+        "x-ms-examples": {
+          "RecommendedActionSessionResult": {
+            "$ref": "./examples/RecommendedActionSessionResult.json"
+          }
+        },
+        "description": "Recommendation action session operation result.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/LocationNameParameter"
+          },
+          {
+            "$ref": "#/parameters/OperationIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/RecommendationActionsResultList"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/RecommendationActionsResultList"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ProxyResource": {
+      "description": "Resource properties.",
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource ID"
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource name."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type."
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "AdvisorProperties": {
+      "properties": {},
+      "description": "The properties of a recommendation action advisor."
+    },
+    "Advisor": {
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/AdvisorProperties",
+          "x-ms-client-flatten": true,
+          "description": "The properties of a recommendation action advisor."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
+        }
+      ],
+      "description": "Represents a recommendation action advisor."
+    },
+    "AdvisorsResultList": {
+      "description": "A list of query statistics.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The list of recommendation action advisors.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Advisor"
+          },
+          "readOnly": true
+        },
+        "nextLink": {
+          "description": "Link to retrieve next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "RecommendationActionProperties": {
+      "properties": {
+        "advisorName": {
+          "type": "string",
+          "description": "Advisor name."
+        },
+        "sessionId": {
```

Comment:
Please go through the models in all these files and mark properties that are "readOnly" as such with "readOnly": true.